pkg/apis/tenancy/v1alpha1/types.go

```diff
@@ -117,6 +135,21 @@ type ClusterWorkspaceTypeSpec struct {
 	//
 	// +optional
 	AdditionalWorkspaceLabels map[string]string `json:"additionalWorkspaceLabels,omitempty"`
+
+	// defaultSubWorkspaceType is the ClusterWorkspaceType that will be used
+	// by default if another, nested ClusterWorkspace is created in a workspace
+	// of this type. The default behaviour requires the user to specify a type.
+	//
+	// +optional
+	DefaultSubWorkspaceType string `json:"defaultSubWorkspaceType,omitempty"`
+
+	// allowedSubWorkspaceTypes is a list of ClusterWorkspaceTypes that can be
+	// created in a workspace of this type.
+	//
+	// By default, no type is allowed. This means no other workspace can be nested
+	// within a workspace of the given type. The name `*` allows any child type to
+	// be nested.
+	AllowedSubWorkspaceTypes []string `json:"allowedSubWorkspaceTypes,omitempty"`
```

Comment:
Assuming `AllowedParentTypes` is coming in follow-up? That would be much more natural for Org and Team.